metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -386,6 +454,125 @@ def _populate_usage_stat_registry(self):
             self.tableau_stat_registry[view.id] = UsageStat(view_count=view.total_views)
         logger.debug("Tableau stats %s", self.tableau_stat_registry)
 
+    def _get_all_project(self) -> Dict[str, TableauProject]:
+        all_project_map: Dict[str, TableauProject] = {}
+
+        def fetch_projects():
+            if self.server is None:
+                return all_project_map
+
+            for project in TSC.Pager(self.server.projects):
+                all_project_map[project.id] = TableauProject(
+                    id=project.id,
+                    name=project.name,
+                    parent_id=project.parent_id,
+                    description=project.description,
+                    path=[],
+                )
+
+        def set_project_path():
+            def form_path(project_id: str) -> List[str]:
+                cur_proj = all_project_map[project_id]
+                ancestors = [cur_proj.name]
+                while cur_proj.parent_id is not None:
+                    cur_proj = all_project_map[cur_proj.parent_id]
+                    ancestors.insert(0, cur_proj.name)
+                return ancestors
+
+            for project in all_project_map.values():
+                project.path = form_path(project_id=project.id)
+                logger.debug(f"Project {project.name}({project.id})")
+                logger.debug(f"Project path = {project.path}")
+
+        fetch_projects()
+        set_project_path()
+
+        return all_project_map
+
+    def _is_allowed_project(self, project: TableauProject) -> bool:
+        # Either project name or project path should exist in allow
+        is_allowed: bool = self.config.project_pattern.allowed(
+            project.name
+        ) or self.config.project_pattern.allowed(self._get_project_path(project))
+        if is_allowed is False:
```

Comment:
What if there is no project pattern provided? Then we should allow all right?

Reply:
As per current behaviour of tableau source the project "default" would get ingested.

Reply:
is that sufficient? should we just allow all by default? feels like maybe we should